cleverpush/src/main/java/com/cleverpush/util/BroadcastReceiverUtils.java

```diff
@@ -0,0 +1,92 @@
+     *
+     * @param context
+     * @function registerReceiver
+     */
+    public static void registerReceiver(CleverPush context) {
+        if (context.registeredDeviceBroadcastReceiver) {
+            return;
+        }
+
+        context.getChannelConfig(channelConfig -> {
+            if (channelConfig.optString(Constants.DEVICE_ID_CONFIG_FIELD) == null || channelConfig.optBoolean(Constants.DEVICE_ID_CONFIG_FIELD) != true) {
+                return;
+            }
+
+            context.registeredDeviceBroadcastReceiver = true;
+            IntentFilter intentFilter = new IntentFilter();
+            intentFilter.addAction(Constants.DEVICE_ID_ACTION_KEY);
+            intentFilter.addAction(Constants.GET_DEVICE_ID_FROM_ALL_DEVICE);
+            CleverPush.context.registerReceiver(broadcastReceiverHandler, intentFilter);
+
+            SharedPreferences sharedPreferences = getSharedPreferences(CleverPush.context);
+            if (sharedPreferences.getString(CleverPushPreferences.DEVICE_ID, null) == null) {
+                String uid = UUID.randomUUID().toString();
+                sharedPreferences.edit().putString(CleverPushPreferences.DEVICE_ID, uid).apply();
+            }
+
+            getDeviceIdFromOtherApps();
+        });
+    }
+
+    /**
+     * This function will send an event to other apps which inform receivers to send their stored device id if they have any
+     *
+     * @function getDeviceIdFromOtherApps
+     */
+    public static void getDeviceIdFromOtherApps() {
+        final Intent intent = new Intent();
+        intent.setAction(Constants.GET_DEVICE_ID_FROM_ALL_DEVICE);
+        intent.putExtra(Constants.GET_FULL_PACKAGE_NAME_KEY, Constants.APPLICATION_PACKAGE_NAME);
+        intent.addFlags(Intent.FLAG_INCLUDE_STOPPED_PACKAGES);
+        context.sendBroadcast(intent);
+    }
+
+    /**
+     * This function will send Device id from shared preference to all other applications
+     *
+     * @function sendBroadcastMessage
+     */
+    public static void sendBroadcastMessage() {
+        SharedPreferences sharedPreferences = getSharedPreferences(context);
+        String deviceId = sharedPreferences.getString(CleverPushPreferences.DEVICE_ID, null);
+
+        final Intent intent = new Intent();
+        intent.setAction(Constants.DEVICE_ID_ACTION_KEY);
+        intent.putExtra(Constants.DEVICE_ID, deviceId);
+        intent.putExtra(Constants.GET_FULL_PACKAGE_NAME_KEY, Constants.APPLICATION_PACKAGE_NAME);
+        intent.addFlags(Intent.FLAG_INCLUDE_STOPPED_PACKAGES);
+        context.sendBroadcast(intent);
+    }
+
+    /**
+     * This function will return Shared preference from preference manager
+     *
+     * @param context
+     * @return SharedPreference
+     * @function getSharedPreferences
+     */
+    public static SharedPreferences getSharedPreferences(Context context) {
```

Comment:
good idea but shouldnt be inside of broadcast file since it works without any broadcast context

Reply:
![image](https://user-images.githubusercontent.com/104719696/195085405-1e37b07b-4f38-481b-9400-8e5abba74748.png)
Yes that's a good idea and I added it like that and removed it below this line
` public static SharedPreferences getSharedPreferences(Context context) {`